src/name2dim.jl

```diff
@@ -0,0 +1,47 @@
+
+function NamedNotFoundException(name, cands)
+    return ArgumentError(
+        "No dimensioned called $name exists. Dimension names: $(join(cands, ", "))"
+    )
+end
+
+###################################
+
+# We change the name into a `Val` because that will trigger constant propergation
+# and allow this to resolve at compile time when the `name` is a constant
+name2dim(namemap, name) = _name2dim(namemap, Val(name))
```

Comment:
If you have a trivial NamedTuple where names point to their position in the tuple:
```julia
dims = (foo = 1, bar = 2)
```
Then this function will do what `name2dim` does at compile time if possible:
```
name2dim(dims_nt, name) = first(merge(NamedTuple{(name,), Tuple{Int}}((0,)), dims_nt))
```